### ISSUE TITLE
eventsByTag querying strategy

### ISSUE THREAD

**Issue by robot-govorun** [dnvriend/akka-persistence-jdbc#149]
Hi everyone!

I've noticed that, when trying to query events by tag which is not present in journal from `given offset` the ReadJournal may run quite beefy queries against the database. Load grows while journal size increases, if the events are at the beginning of journal (before `given offset`). After some research I figured out that, when the ReadJournal queries `eventsByTag` with `terminateAfterOffset` = None, empty result set leads to scheduling next query iteration with query condition to start with `given offset` but the previous `max ordering id`. The difference between `given offset` and `max ordering id` continuously increases because the number of events of no interest may grow, which makes further queries more and more heavy (in my case the difference may be up to 1kk). Of course, this happens until event of interest appears at the tail. 

It may look like a bug, what do you think about that?

**Comment by WellingR** [comment]
The eventsByTag query has been implemented to query for all new events, from the last known event which matched the given tags.

Because of two bugs (#96 and #124) some changes to the query algorithm were made. We now figure out the highest ordering value (`maxOrderingId`) for which no gap appear in the journal. And then we query all events until that value.

Since we know exactly from what id and until what id we have queried. I think you are right that we actually can increase the from value. So that less events potentially match the query.

Thanks for noticing this. You are right that this might cause unnecessary database load. I will have a look if we can improve this.
